src/components/Nav/index.tsx

```diff
@@ -92,16 +94,42 @@ const Nav: FC<IProps> = ({ path }) => {
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    transform: "rotate(30deg)", // Rotate the icon to 30 degrees on hover
+                    color: "primary.hover", // Change color to blue on hover
+                  }}
                   onClick={toggleColorMode}
                 ></IconButton>
+
                 <ButtonLink
+                  onMouseOver={() => setLanguagesHover(true)}
+                  onMouseOut={() => setLanguagesHover(false)}
                   to={`/languages/${fromPageParameter}`}
-                  leftIcon={<Icon as={MdLanguage} />}
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    color: "primary.hover", // Change color on hover
+                  }}
                 >
-                  {t("languages")} {i18n.language.toUpperCase()}
+                  <Box mr={2} mt={2}>
+                    {" "}
+                    {/* Add spacing between the text and the icon */}
+                    <ButtonLink
+                      data-group
+                      leftIcon={
```

Comment:
Not sure if I follow what we are trying to do here but currently, you have a nested `ButtonLink` inside another `ButtonLink`, resulting in a nested `a` tag.

I guess that what we would like to do here is to render just the icon, right? and add the `data-group` tag on the parent `ButtonLink`.

====
Note aside, the build is failing because this `ButtonLink` is missing the `href` prop.

Reply:
Yea, the example I gave was not suggesting nested, but a simplified version of what the component should look like with that `data-*` prop and the `leftIcon` prop